File: PROGRAMMERS/올바른괄호.py
def solution(s):
    answer = True
    stk = []

    for x in s:
        if x == "(":
            stk.append(x)
        else:
            if stk == []:
                answer = False
                break
            else:
                stk.pop()
    if stk != []:
        answer = False

    return answer


def solution(s):
    answer = True
    cnt = 0
    if s[0] == ")":
        return False
    if s[-1] == "(":
        return False
    else:
        for x in s:
            if x == "(":
                cnt += 1
            elif x == ")":
                cnt -= 1
                if cnt < 0:
                    return False

    if cnt != 0:
        answer = False
    return answer

File: PROGRAMMERS/JAVA/src/PG/Lv_2/올바른괄호.java
package PG.Lv_2;

import java.util.Stack;

public class 올바른괄호 {
    static boolean solution(String s) {
        boolean answer = true;
        Stack<String> stk = new Stack<>();

        for (int i = 0; i < s.length(); i++) {
            if (s.charAt(i) == '(') {
                stk.push("(");
            } else {
                if (stk.isEmpty()) {
                    return false;
                } else {
                    stk.pop();
                }
            }
        }
        if (!stk.isEmpty()) {
            return false;
        }
        return answer;
    }

    public static void main(String[] args) {
        boolean result = solution("()()");
        System.out.println(result);
    }
}
